x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java

```diff
@@ -341,17 +342,18 @@ public StreamInput streamInput() throws IOException {
                     };
                     if (atomic) {
                         try {
-                            blobContainer.writeBlobAtomic(request.blobName, bytesReference, failIfExists);
+                            blobContainer.writeBlobAtomic(OperationPurpose.SNAPSHOT, request.blobName, bytesReference, failIfExists);
```

Comment:
We could reasonably distinguish the purpose of the repo analysis calls (in a followup tho)